mathesar_ui/src/utils/languageUtils.ts

```diff
@@ -110,12 +114,53 @@ export function pluralize(
   }
 }
 
+type CountWhenZero = 'numeric' | { word: string };
+type CountWhenSingular = 'hidden' | 'numeric' | { word: string };
+
+function getCountText(
+  count: number,
+  countWhenZero: CountWhenZero,
+  countWhenSingular: CountWhenSingular,
+): string {
+  if (count === 0) {
+    if (countWhenZero === 'numeric') {
+      return String(count);
+    }
+    if ('word' in countWhenZero) {
```

Comment:
I generally prefer `hasOwnProperty`(or its latest incarnation: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/hasOwn) instead of `in` as a convention to avoid looking into prototype chains un-intentionally.

Reply:
Interesting 🤔

- Thanks for pointing out `Object.hasOwn`. I had not seen that before. It looks nice, although I'd say [support](https://caniuse.com/mdn-javascript_builtins_object_hasown) is still not sufficient that we should use it in our codebase (without a polyfill).

- I've not seen `hasOwnProperty` used for type narrowing before.

    If I change the code to use what I _think_ you're suggesting, then:

    - I get this ESLint error.

        ![image](https://user-images.githubusercontent.com/42411/212129507-b6f676f4-7ac2-40af-abf8-c11fd24b30d1.png)

        We could silence that linting error if needed. But...

    - TypeScript does not seem to actually narrow the type based on that code

        ![image](https://user-images.githubusercontent.com/42411/212129839-df4fee9c-7030-4e25-87d9-4b8b7492b861.png)

    From the TypeScript code I've seen, usage of `in` seems to be idiomatic. Indeed, that's how the [TypeScript docs](https://www.typescriptlang.org/docs/handbook/2/narrowing.html#the-in-operator-narrowing) recommend narrowing types. That docs page has no occurrences of `hasOwnProperty`.

    Am I'm misunderstanding your recommendation here?

Reply:
@seancolsen its not about type narrowing - we can acheive this on `Object.hasOwnProperty` by overriding its type - rather about avoid looking into prototype chains. Anyways its a small thing hence approving the PR

Reply:
I'd be curious to see the code you'd prefer to use in this case.